File: python/scrust/metrics/_autocorrelation.py
"""Spatial autocorrelation over the neighbour graph. Owned by feat/metrics."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    import numpy as np
    from anndata import AnnData

__all__ = ["gearys_c", "morans_i"]


def gearys_c(
    adata: AnnData, *, vals: Any = None, use_graph: str = "connectivities", device: str = "auto"
) -> np.ndarray:
    """Geary's C for each gene over the neighbour graph, as `scanpy.metrics.gearys_c`."""
    raise NotImplementedError("feat/metrics")


def morans_i(
    adata: AnnData, *, vals: Any = None, use_graph: str = "connectivities", device: str = "auto"
) -> np.ndarray:
    """Moran's I for each gene over the neighbour graph, as `scanpy.metrics.morans_i`."""
    raise NotImplementedError("feat/metrics")

File: crates/scrust-core/src/autocorrelation.rs
//! Spatial autocorrelation over the neighbour graph. Owned by feat/metrics.

use candle_core::Device;

use crate::error::Result;
use crate::sparse::CsrMatrix;

/// Moran's I for every column of `values`, as `scanpy.metrics.morans_i`.
pub fn morans_i(_graph: &CsrMatrix, _values: &CsrMatrix, _device: &Device) -> Result<Vec<f32>> {
    todo!("feat/metrics")
}

/// Geary's C for every column of `values`, as `scanpy.metrics.gearys_c`.
pub fn gearys_c(_graph: &CsrMatrix, _values: &CsrMatrix, _device: &Device) -> Result<Vec<f32>> {
    todo!("feat/metrics")
}
